src/jemalloc.c

```diff
@@ -1158,27 +1159,45 @@ malloc_conf_init_helper(sc_data_t *sc_data, unsigned bin_shard_sizes[SC_NBINS],
 #define CONF_HANDLE_SIZE_T(o, n, min, max, check_min, check_max, clip)	\
 			CONF_HANDLE_T_U(size_t, o, n, min, max,		\
 			    check_min, check_max, clip)
-#define CONF_HANDLE_SSIZE_T(o, n, min, max)				\
+#define CONF_HANDLE_T_SIGNED(t, o, n, min, max, check_min, check_max, clip)	\
 			if (CONF_MATCH(n)) {				\
-				long l;					\
+				intmax_t im;				\
 				char *end;				\
 									\
 				set_errno(0);				\
-				l = strtol(v, &end, 0);			\
+				im = (intmax_t)malloc_strtoumax(v, &end, 0);\
```

Comment:
I like that we extract out common logic for all signed integers.

Question: what if the input is `-1`? I don't think we have a `malloc_strtomax()`. I looked into `malloc_strtoumax()` and it looks like it would eventually return `UINTMAX_MAX` for a `-1` input, so when we cast it back to `intmax_t` it will go back to `-1`. So correctness wise it's fine, though we may want to add some comments. Meanwhile, that makes me wonder, can the existing `CONF_HANDLE_T_U` serve the purpose of this new `CONF_HANDLE_T_SIGNED`?

Reply:
Yeah will add comments.

re: the unsigned version VS this one, I don't think we can easily reuse the unsigned one; the comparison would be messed up (i.e. comparing the negative values with unsigned types).

Reply:
I think it should work if you do the typed cast immediately on the result returned from `malloc_strtoumax()`? Afterwards the min/max comparisons should be fine.

Reply:
I think you are probably right. Let me give it a try.

Reply:
Merged the two marcos.